src/NuGet.Clients/NuGet.VisualStudio.OnlineEnvironment.Client/SolutionExplorer/NuGetWorkspaceCommandHandler.cs

```diff
@@ -65,24 +79,64 @@ public bool QueryStatus(List<WorkspaceVisualNodeBase> selection, Guid pguidCmdGr
             if (pguidCmdGroup == CommandGroup.NuGetOnlineEnvironmentsClientProjectCommandSetGuid)
             {
                 var nCmdIDInt = (int)nCmdID;
-
-                if (IsSolutionOnlySelection(selection))
                 {
                     switch (nCmdIDInt)
                     {
                         case PkgCmdIDList.CmdidRestorePackages:
-                            var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
-                            cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
-                                | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
-                            handled = true;
+                            if (IsSolutionOnlySelection(selection))
+                            {
+
+                                var isRestoreActionInProgress = _restoreCommandHandler.IsRestoreActionInProgress();
+                                cmdf = (uint)((isRestoreActionInProgress ? 0 : Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED)
+                                    | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED);
+                                handled = true;
+                            }
+                            break;
+                        case PkgCmdIDList.CmdIdManageProjectUI:
+                            if (IsSupportedProjectSelection(selection))
+                            {
+                                var isPackageManagerUISupported = _packageManagerUICommandHandler.IsPackageManagerUISupported(selection.Single());
+                                cmdf = (uint)(isPackageManagerUISupported ?
+                                    (Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_ENABLED | Microsoft.VisualStudio.OLE.Interop.OLECMDF.OLECMDF_SUPPORTED) :
```

Comment:
nit: try to add some using statements to avoid using fully qualified name of the constant

Reply:
fixed.